datafusion/datasource-avro/src/avro_to_arrow/arrow_array_reader.rs

```diff
@@ -359,13 +355,15 @@ impl<R: Read> AvroArrowArrayReader<'_, R> {
                         let builder = builder.as_any_mut().downcast_mut::<ListBuilder<StringDictionaryBuilder<D>>>().ok_or_else(||SchemaError(
                             "Cast failed for ListBuilder<StringDictionaryBuilder> during nested data parsing".to_string(),
                         ))?;
-                        for val in vals {
+                        vals.into_iter().try_for_each(|val| {
                             if let Some(v) = val {
-                                let _ = builder.values().append(&v)?;
+                                builder.values().append(v)?;
                             } else {
-                                builder.values().append_null()
-                            };
-                        }
+                                builder.values().append_null();
+                            }
+
+                            Result::<_, ArrowError>::Ok(())
```

Comment:
This looks fishy.
`Result` is `datafusion_common::error::Result` https://github.com/EmilyMatt/datafusion/blob/0ea75919a3b373b6719601cd69474d14f377e319/datafusion/datasource-avro/src/avro_to_arrow/arrow_array_reader.rs#L49
 but it is a type alias with `E = DataFusionError` https://github.com/apache/datafusion/blob/0ea75919a3b373b6719601cd69474d14f377e319/datafusion/common/src/error.rs#L61
How does this compile ?!